### PR TITLE
[DT-336] Profile View Refactor

### PR DESCRIPTION
**Pull request by AngeloMendonca**
**What does this PR do?**

   Refactor the profile view following the settings view structure

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] by commits

**How should this be manually tested?**

  Flow on how to test this or QA Tickets related to this use-case: [DT-336](https://aptoide.atlassian.net/browse/DT-336)

**What are the relevant tickets?**

  Tickets related to this pull-request: [DT-336](https://aptoide.atlassian.net/browse/DT-336)

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

[DT-336]: https://aptoide.atlassian.net/browse/DT-336?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DT-336]: https://aptoide.atlassian.net/browse/DT-336?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt [hunk 1]
```diff
@@ -0,0 +1,537 @@
...
+        .clickable(onClick = editImageOnClick)
+        .size(64.dp)
+        .clip(CircleShape)
+        .border(
+          width = 4.dp,
+          color = AppTheme.colors.imageIconBackground,
+          shape = CircleShape,
+        )
+    )
+
```
Comment (G0retZ) [inline feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt:166]: unnecessary new line

File: feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt [hunk 2]
```diff
@@ -0,0 +1,537 @@
...
+      Text(
+        text = "Joined Aptoide in $userJoinedData",
+        modifier = Modifier.fillMaxWidth(),
+        textAlign = TextAlign.Start,
+        overflow = TextOverflow.Ellipsis,
+        maxLines = 1,
+        style = AppTheme.typography.medium_XXS
+      )
+    }
+
```
Comment (G0retZ) [inline feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt:192]: unnecessary new line

File: feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt [hunk 3]
```diff
@@ -0,0 +1,537 @@
...
+    imageVector = AppTheme.icons.AptoideIcon,
+    titleColor = AppTheme.colors.onBackground,
+    imageColorFilter = null,
+    contentDescription = "AptoideIcon",
+  ) {
+    Column(
+      modifier = Modifier.fillMaxWidth(),
+      verticalArrangement = Arrangement.spacedBy(0.dp)
+    ) {
+
```
Comment (G0retZ) [inline feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt:254]: unnecessary new line

File: feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt [hunk 4]
```diff
@@ -0,0 +1,537 @@
...
+      ProductSection(
+        title = "Aptoide TV",
+        description = "The best solution for your Set Top Box and Smart TV",
+        imageVector = AppTheme.icons.AptoideTVIcon,
+        contentDescription = "Aptoide TV",
+        onClick = { openLinkFunc("https://tv.aptoide.com/") }
+      )
+
```
Comment (G0retZ) [inline feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt:262]: unnecessary new line

File: feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt [hunk 5]
```diff
@@ -0,0 +1,537 @@
...
+      ProductSection(
+        title = "Aptoide Uploader",
+        description = "Perfect tool to get your favorite apps in Aptoide store",
+        imageVector = AppTheme.icons.AptoideUploaderIcon,
+        contentDescription = "Aptoide Uploader",
+        onClick = { openLinkFunc("https://aptoide-uploader.pt.aptoide.com/app") }
+      )
+
```
Comment (G0retZ) [inline feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt:270]: unnecessary new line

File: feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt [hunk 6]
```diff
@@ -0,0 +1,537 @@
...
+      Row(
+        modifier = Modifier
+          .fillMaxWidth()
+          .padding(start = 16.dp, top = 6.dp),
+        horizontalArrangement = Arrangement.spacedBy((-4).dp),
+        verticalAlignment = Alignment.CenterVertically,
+
```
Comment (G0retZ) [inline feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt:277]: unnecessary new line

File: feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt [hunk 7]
```diff
@@ -0,0 +1,537 @@
...
+        ) {
+        Text(
+          text = "Follow us",
+          modifier = Modifier.padding(end = 12.dp),
+          textAlign = TextAlign.Start,
+          overflow = TextOverflow.Ellipsis,
+          maxLines = 1,
+          style = AppTheme.typography.medium_M
+        )
+
```
Comment (G0retZ) [inline feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt:287]: unnecessary new line

File: feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt [hunk 8]
```diff
@@ -0,0 +1,537 @@
...
+        FollowUsIcon(
+          imageVector = AppTheme.icons.FacebookIcon,
+          contentDescription = "FacebookIcon",
+          onClick = {
+            openLinkFunc(
+              "https://www.facebook.com/aptoide"
+            )
+          }
+        )
+
```
Comment (G0retZ) [inline feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt:297]: unnecessary new line

File: feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt [hunk 9]
```diff
@@ -0,0 +1,537 @@
...
+        FollowUsIcon(
+          imageVector = AppTheme.icons.TwitterIcon,
+          contentDescription = "TwitterIcon",
+          onClick = {
+            openLinkFunc(
+              "https://twitter.com/Aptoide"
+            )
+          }
+        )
+
```
Comment (G0retZ) [inline feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt:307]: unnecessary new line

File: feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt [hunk 10]
```diff
@@ -0,0 +1,537 @@
...
+        horizontalArrangement = Arrangement.spacedBy(16.dp)
+      ) {
+        Image(
+          imageVector = imageVector,
+          colorFilter = imageColorFilter,
+          contentDescription = contentDescription,
+          contentScale = ContentScale.Fit,
+          modifier = Modifier.size(28.dp)
+        )
+
```
Comment (G0retZ) [inline feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt:438]: unnecessary new line

File: feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt [hunk 11]
```diff
@@ -0,0 +1,537 @@
...
+        Text(
+          text = title,
+          modifier = Modifier.weight(1f),
+          textAlign = TextAlign.Start,
+          overflow = TextOverflow.Ellipsis,
+          maxLines = 1,
+          style = AppTheme.typography.medium_M,
+          color = titleColor
+        )
+
```
Comment (G0retZ) [inline feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt:448]: unnecessary new line

File: feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt [hunk 12]
```diff
@@ -0,0 +1,537 @@
...
+      Image(
+        imageVector = imageVector,
+        contentDescription = contentDescription,
+        contentScale = ContentScale.Fit,
+        modifier = Modifier
+          .wrapContentHeight()
+          .size(50.dp)
+          .clip(RoundedCornerShape(5.dp))
+      )
+
```
Comment (G0retZ) [inline feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt:487]: unnecessary new line

File: feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt [hunk 13]
```diff
@@ -0,0 +1,537 @@
...
+          style = AppTheme.typography.medium_XS,
+          color = grey
+        )
+      }
+    }
+  }
+}
+
+@Composable
+private fun FollowUsIcon(
```
Comment (G0retZ) [inline feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt:514]: Function name is misleading

**Review by G0retZ — CHANGES_REQUESTED** [review]

File: feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt [hunk 14]
```diff
@@ -0,0 +1,537 @@
...
+      modifier = Modifier
+        .wrapContentWidth()
+        .clickable(onClick = editImageOnClick),
+      textAlign = TextAlign.Start,
+      overflow = TextOverflow.Ellipsis,
+      maxLines = 1,
+      style = AppTheme.typography.medium_XS,
+      color = pinkishOrange
+    )
+    Spacer(modifier = Modifier.width(1.dp))
```
[the remark this answers is not in the document]
Comment (AngeloMendonca) [reply feature-profile/src/main/java/cm/aptoide/pt/profile/presentation/ProfileView.kt]: For no reason, i don't know why i used it in the first place xD

**Review by G0retZ — APPROVED** [review]

**Pull request merged**